### PR TITLE
[minigraph.py]: Add support for parsing mux cable

### PR DESCRIPTION
**Pull request by theasianpianist**
Signed-off-by: Lawrence Lee <lawlee@microsoft.com>

**- Why I did it**

**- How I did it**
Find LogicalLinks in minigraph and parse the port information. A new field called `mux_cable` is added to each port's entry in the Port table in config DB:

```
PORT|Ethernet0: {
	"alias": "Ethernet4/1"
	...
	"mux_cable": "true"
}
```

If a mux cable is present on a port, the value for `mux_cable` will be `"true"`. If no mux cable is present, the attribute will either be omitted (default behavior) or set to `"false"`.

**- How to verify it**
Run sonic-config-engine tests

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: src/sonic-config-engine/tests/test_minigraph_case.py [hunk 1]
```diff
@@ -159,3 +160,13 @@ def test_minigraph_bgp_mon(self):
...
+    def test_mux_cable_parsing(self):
+        result = minigraph.parse_xml(self.sample_graph, port_config_file=self.port_config)
+        
+        expected_mux_cable_ports = {"Ethernet4":"l", "Ethernet8":"u"}
+        port_table = result['PORT']
+        for port_name, port in port_table.items():
+            if port_name in expected_mux_cable_ports:
+                self.assertEqual(port["mux_cable_connection"], expected_mux_cable_ports[port_name])
+            else:
+                self.assertTrue("mux_cable_connection" not in port)
```
Comment (jleveque) [inline src/sonic-config-engine/tests/test_minigraph_case.py:172]: Add newline at EOF

**Review by jleveque — CHANGES_REQUESTED** [review]

**Comment by lguohan** [comment]
do we need to differentiate U and L for a mux cable port? what is the motivation here?

**Comment by theasianpianist** [comment]
@lguohan the minigraph will list the `<StartPort>` as either U or L to distinguish the upper and lower ToRs. We can change this field to true/false and move upper/lower ToR info to somewhere else if necessary.

@vdahiya12 for visibility

**Comment by theasianpianist** [comment]
retest broadcom please

**Comment by vdahiya12** [comment]
> @lguohan the minigraph will list the `<StartPort>` as either U or L to distinguish the upper and lower ToRs. We can change this field to true/false and move upper/lower ToR info to somewhere else if necessary.
> 
> @vdahiya12 for visibility

@lguohan Can the Cli request to change the active TOR to be U or L, if not we can omit this ?

**Review by vdahiya12 — APPROVED** [review]

**Review by jleveque — APPROVED** [review]

**Comment by lguohan** [comment]
retest vsimage please

**Review by vdahiya12 — APPROVED** [review]

**Review by lguohan — APPROVED** [review]

**Pull request merged**